Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionVMwareCarbonBlackCloud.yaml

```diff
@@ -0,0 +1,208 @@
+Parser:
+  Title: NetworkSession ASIM Parser for VMware Carbon Black Cloud
+  Version: '0.1.0'
+  LastUpdated: Oct 04, 2023
+Product:
+  Name: VMware Carbon Black Cloud
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.6'
+References:
+- Title: ASIM NetworkSession Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: VMware Carbon Black Cloud documentation 
+  Link: https://developer.carbonblack.com/reference/carbon-black-cloud/
+        https://developer.carbonblack.com/reference/cb-threathunter/latest/event-search-fields/
+Description: |
+  This ASIM parser supports normalizing VMware Carbon Black Cloud logs to the ASIM NetworkSession normalized schema. VMware Carbon Black Cloud events are captured through VMware Carbon Black Cloud data connector which ingests Carbon Black Audit, Notification and Event data into Microsoft Sentinel through the REST API.
+ParserName: ASimNetworkSessionVMwareCarbonBlackCloud
+EquivalentBuiltInParser: _ASim_NetworkSession_VMwareCarbonBlackCloud
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let NetworkProtocolLookup = datatable (netconn_protocol_s: string, NetworkProtocol: string)
+  [
+      "PROTO_TCP", "TCP",
+      "PROTO_UDP", "UDP"
+  ];
+  let DvcActionLookup = datatable (sensor_action_s: string, DvcAction: string)
+      [
+      "ACTION_ALLOW", "Allow",
+      "ACTION_SUSPEND", "Drop",
+      "ACTION_TERMINATE", "Drop",
+      "ACTION_BREAK", "Drop",
+      "ACTION_BLOCK", "Deny"
+  ];
+  let EventSeverityLookup = datatable (DvcAction: string, EventSeverity: string)
+      [
+      "Allow", "Informational",
+      "Drop", "Low",
+      "Deny", "Low"
+  ];
+  let ThreatConfidenceLookup = datatable (ThreatOriginalConfidence: string, ThreatConfidence: int)
+      [
+      "1", 10,
+      "2", 20,
+      "3", 30,
+      "4", 40,
+      "5", 50,
+      "6", 60,
+      "7", 70,
+      "8", 80,
+      "9", 90,
+      "10", 100
+  ];
+  let parser=(disabled: bool=false) {
+      let alldata = CarbonBlackEvents_CL
+          | where not(disabled)
+          | where eventType_s == "endpoint.event.netconn"
+          | lookup NetworkProtocolLookup on netconn_protocol_s
+          | lookup DvcActionLookup on sensor_action_s
+          | lookup EventSeverityLookup on DvcAction;
+      let threatsdata = CarbonBlackNotifications_CL
+          | where type_s == "THREAT"
+          | project
+              threatInfo_incidentId_g,
+              threatInfo_score_d,
+              threatInfo_summary_s,
+              threatInfo_time_d,
+              threatInfo_threatCause_threatCategory_s,
+              threatInfo_threatCause_causeEventId_g,
+              ruleName_s,
```

Comment:
Is this relevant with NetworkRuleName?

Reply:
This is the Notification rule which has generated notification for this alert containing threat information.